File: c atau c++/ATM.cpp
#include <iostream>
using namespace std;

int main() {
    string nama, bank; 
    char konfirmasi; 
    int nim, saldo, menu, jumlah, norek;

    do {
        cout << "Masukkan Nama : ";
        cin >> nama;

        cout << "Apakah nama anda sudah benar? (Y/N)  ";
        cin >> konfirmasi;

    } while (konfirmasi == 'N' || konfirmasi == 'n');

    cout << "Masukkan NIM : ";
    cin >> nim;
    saldo = nim;

    cout << "saldo awal anda : Rp " << saldo << endl;

    do {
        cout << "\n========== MWNU ATM ==========\n";
        cout << "1. Cek Saldo\n";
        cout << "2. Tarik Tunai\n";
        cout << "3. Setor Tunai\n";
        cout << "4. Transfer\n";
        cout << "5. Keluar\n";
        cout << "Menu ";
        cin >> menu;

        switch(menu) {

            case 1:
                cout << "Saldo anda : Rp " << saldo << endl;
                break;

            case 2:
                cout << "Jumlah saldo yang ingin ditarik : ";
                cin >> jumlah;

                if (jumlah > saldo) {
                    cout << "Saldo tidak cukup, tidak dapat ditarik\n";
                } else {
                    saldo -= jumlah;
                    cout << "berhasil. Saldo Sekarang : Rp " << saldo << endl;
                }
                break;

            case 3:
                cout << "Jumlah setor tunai : ";
                cin >> jumlah;
                saldo += jumlah;
                cout << "berhasil. Saldo sekarang : Rp " << saldo << endl;
                break;

            case 4: 
                cout << "Bank tujuan: ";
                cin >> bank;

                cout << "No Rekening: ";
                cin >> norek;
                cout << "Jumlah yang akan ditransfer : ";
                cin >> jumlah;

                if (jumlah > saldo) {
                    cout << "Saldo tidak cukup, tidak dapat transfer\n";
                } else {
                    saldo -= jumlah;
                    cout << "berhasil. saldo sekarang : Rp " << saldo << endl;
                }
                break;

            case 5:
                cout << "terimakasih\n";
                break;

            default:
                cout << "Menu tidak valid!\n";
        }

    } while (menu != 5);

    return 0;
}

File: python/ATM.py
def validasi_nama(nama):
    return nama.isalpha()

while True:
    nama = input("Masukkan nama: ")
    if validasi_nama(nama):
        break
    else:
        print("Nama tidak valid, coba lagi!")

nim = input("Masukkan NIM: ")
saldo = int(nim)

print(f"\nHalo {nama}, saldo kamu adalah Rp {saldo:,}")

while True:
    print("\n===== MENU ATM =====")
    print("1. Cek Saldo")
    print("2. Tarik Tunai")
    print("3. Setor Tunai")
    print("4. Transfer")
    print("5. Keluar")

    pilih = input("Pilih menu: ")

    if pilih == "1":
        print(f"Saldo kamu: Rp {saldo:,}")
    elif pilih == "2":
        tarik = int(input("Jumlah tarik: "))
        if tarik <= saldo:
            saldo -= tarik
            print("Penarikan berhasil.")
        else:
            print("Saldo tidak cukup!")
    elif pilih == "3":
        setor = int(input("Jumlah setor: "))
        saldo += setor
        print("Setor berhasil.")
    elif pilih == "4":
        tujuan = input("Rekening tujuan: ")
        nominal = int(input("Nominal transfer: "))
        if nominal <= saldo:
            saldo -= nominal
            print(f"Transfer ke {tujuan} berhasil.")
        else:
            print("Saldo tidak cukup!")
    elif pilih == "5":
        print("Terima kasih!")
        break
    else:
        print("Menu tidak valid!")